web/store/issue/module/issue.store.ts

```diff
@@ -359,24 +370,37 @@ export class ModuleIssues extends IssueHelperStore implements IModuleIssues {
 
   addModulesToIssue = async (workspaceSlug: string, projectId: string, issueId: string, moduleIds: string[]) => {
     try {
-      const issueToModule = await this.moduleService.addModulesToIssue(workspaceSlug, projectId, issueId, {
-        modules: moduleIds,
-      });
-
       runInAction(() => {
         moduleIds.forEach((moduleId) => {
+          // add the new issue ids to the module issues map
           update(this.issues, moduleId, (moduleIssueIds = []) => {
             if (moduleIssueIds.includes(issueId)) return moduleIssueIds;
             else return uniq(concat(moduleIssueIds, [issueId]));
           });
         });
+        // update the root issue map with the new module ids
         update(this.rootStore.issues.issuesMap, [issueId, "module_ids"], (issueModuleIds = []) =>
           uniq(concat(issueModuleIds, moduleIds))
         );
       });
 
+      const issueToModule = await this.moduleService.addModulesToIssue(workspaceSlug, projectId, issueId, {
+        modules: moduleIds,
+      });
+
       return issueToModule;
     } catch (error) {
+      moduleIds.forEach((moduleId) => {
+        runInAction(() => {
+          // remove the new issue ids from the module issues map
+          update(this.issues, moduleId, (moduleIssueIds = []) => pull(moduleIssueIds, issueId));
+          // remove the new module ids from the root issue map
+          update(this.rootStore.issues.issuesMap, [issueId, "module_ids"], (issueModuleIds = []) =>
```

Comment:
just a suggestion, can we maybe move this out of for Each and just just use Set to update the "module_ids" of the issue from the array before the update?